### DISCUSSION TITLE
Test Filtering

### DISCUSSION THREAD

**Discussion by jonathanlocke** [xunit/xunit#2794] [Question] [answered]
There seems to be a nice facility in "dotnet test" that allows for exclusion of particular tests from a build. But is there a way to exclude a project in the solution from being tested at all (either through a dotnet feature or through xunit)? The reason I ask is that we have a bizarre testing failure in a project that has no tests, and the easiest thing to do would be to just exclude that project from the solution projects that get tested when we run "dotnet test". I'm really new to C#/.NET, so I probably don't have certain knowledge that you all probably take for granted. Thank you for your time!
Jon

**Comment by bradwilson** [comment, marked as answer]
This feature would have to exist on the `dotnet test` side, and AFAIK it does not exist. You can either run `dotnet test` against a single project, or against a solution (in which case it runs against all projects). I'm not aware of any filtering at that level which says "run against the solution, but not this particular project".

From our perspective, we don't know how/when we get activated to run, so we don't have the necessary context to know what the user requested.

**Reply by jonathanlocke** [reply]
Thanks very much Brad!
